src/Cardano/Wallet/Primitive/Signing.hs

```diff
@@ -0,0 +1,135 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeApplications #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- This module provides functionality for signing transactions.
+--
+-- It relies on the binary CBOR format of transactions,
+module Cardano.Wallet.Primitive.Signing
+    ( -- * Sign transactions
+      mkStdTx
+
+      -- * Helpers
+    , shuffle
+    )
+    where
+
+
+import Prelude
+
+import Cardano.Wallet.Binary
+    ( TxWitness (..), encodeTx, toByteString )
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( ChangeChain (ExternalChain)
+    , Depth (AddressK, RootK)
+    , Key
+    , Passphrase
+    , SignTag (SignTx)
+    , XPrv
+    , deriveAccountPrivateKey
+    , deriveAddressPrivateKey
+    , encodeXPub
+    , publicKey
+    , sign
+    )
+import Cardano.Wallet.Primitive.AddressDiscovery
+    ( SeqState (externalPool), lookupAddress )
+import Cardano.Wallet.Primitive.Types
+    ( Address, Hash (..), Tx (..), TxIn, TxOut (..) )
+import Control.Monad
+    ( forM, forM_ )
+import Crypto.Hash
+    ( Blake2b_256, hash )
+import Crypto.Number.Generate
+    ( generateBetween )
+import Data.ByteArray
+    ( convert )
+import Data.ByteString
+    ( ByteString )
+import Data.Vector.Mutable
+    ( IOVector )
+
+import qualified Data.Vector as V
+import qualified Data.Vector.Mutable as MV
+
+-- | Build a transaction
+
+-- | Construct a standard transaction
+--
+-- " Standard " here refers to the fact that we do not deal with redemption,
+-- multisignature transactions, etc.
+--
+-- TODO: I removed FakeSigner/SafeSigner. Might be wrong.
+mkStdTx :: SeqState
```

Comment:
We ought to support signing for different schemes, so let's also use the abstraction here with `IsOurs s => s ...`

Reply:
We don't need to check if the address is ours. We need to look-up the private key of the address.

I don't see how with `IsOurs`. Should we change/extend the class to 
```haskell
    isOurs :: Address -> s -> (Key 'AddressK XPrv, s)
```
, `Key` instead of `Bool`?

Reply:
Have gone with
```haskell
keyFrom :: Address -> (Key 'Root XPrv, Passphrase "encryption") -> s -> (Maybe (Key 'AddressK XPrv), s)
```
for now as you suggested